clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerClientFactory.java

```diff
@@ -98,29 +100,25 @@ public synchronized <K, V> BootstrappingVeniceChangelogConsumer<K, V> getBootstr
    * Creates a BootstrappingVeniceChangelogConsumer with consumer id. This is used to create multiple
    * consumers so that each consumer can only subscribe to certain partitions.
    */
-  public synchronized <K, V> BootstrappingVeniceChangelogConsumer<K, V> getBootstrappingChangelogConsumer(
+  public <K, V> BootstrappingVeniceChangelogConsumer<K, V> getBootstrappingChangelogConsumer(
       String storeName,
       String consumerId) {
-    String storeClientKey = storeName;
-    if (StringUtils.isNotEmpty(consumerId)) {
-      storeClientKey += "-" + consumerId;
-    }
-
-    return (BootstrappingVeniceChangelogConsumer<K, V>) storeClientMap.computeIfAbsent(storeClientKey, name -> {
-      ChangelogClientConfig newStoreChangelogClientConfig = getNewStoreChangelogClientConfig(storeName);
-      String viewClass = getViewClass(newStoreChangelogClientConfig, storeName);
-      String consumerName = storeName + "-" + viewClass.getClass().getSimpleName();
-      if (StringUtils.isNotEmpty(consumerId)) {
-        consumerName += "-" + consumerId;
-      }
-
-      return new LocalBootstrappingVeniceChangelogConsumer(
-          newStoreChangelogClientConfig,
-          consumer != null
-              ? consumer
-              : getConsumer(newStoreChangelogClientConfig.getConsumerProperties(), consumerName),
-          consumerId);
-    });
+    return (BootstrappingVeniceChangelogConsumer<K, V>) storeClientMap
+        .computeIfAbsent(suffixConsumerIdToStore(storeName, consumerId), name -> {
```

Comment:
can you change `storeClientMap` to `ConcurrentHashMap` so that to guarantee one consumer per client

Reply:
Good catch. I prefer VeniceConcurrentHashMap directly everywhere if possible. Fixed.